kernel/src/platform/chip.rs

```diff
@@ -117,6 +117,9 @@ pub trait ClockInterface {
     fn is_enabled(&self) -> bool;
     fn enable(&self);
     fn disable(&self);
+    fn get_frequency(&self) -> Option<u32> {
```

Comment:
This isn't your fault, but it doesn't make sense that a clock has an optional frequency.

I feel like the issue is a peripheral that doesn't use a clock shouldn't pretend like it does, and the option should be at that level, not trying to fake a clock when one doesn't exist.

I will try to find out if there is a good reason to have fake clocks, and if not I'll update PeripheralManager trait.

Reply:
That would also work.

I would like to avoid adding `.expect()` in multiple places which is conceptually meaningless; there is no valid way for a clock to not have a frequency.

Reply:
I see your point. I can update this PR with an stm32f4xx specific trait where the `get_frequency()` directly returns the `u32` value. However I think such a trait could be useful for other platforms too in the future. Would it make sense to add a separate, but still generic, trait in the chip.rs module for this purpose (e.g. `ClockFrequency`?). It would be implemented only by those clocks who need it.

Reply:
I think this could be useful more generally, but all clocks have a frequency, so I don't see the need for a new trait.

My opinion is either things which are not clocks should just return frequency 0, or getting a clock should return an `Option`.

Reply:
OK, @bradjc I agree. Clocks should always have a frequency and if a peripheral _doesn't_ have a clock, it shouldn't return something of type clock.

I'm not sure it _shouldn't_ be a separate trait, this is only used at all in the sam4l and the msp432. The Sam4l uses it as intended. The MSP432 is the thing that uses the `NoClockControl` hack, and I don't really understand why other than maybe it was a placeholder before completing actual clock control that never got finished.

In any case @matteomartelli I think keeping this _in_ stm for now is a better way forward. More actual use will help inform a better general interface design.

Reply:
Updated the PR removing the commit on the generic `ClockInterface` and moving the `get_frequency()` method inside the strm32f4 `PeripheralClock` specific implementation. I followed the same approach already used by the [trng.rs module](https://github.com/tock/tock/blob/7bfde3e3dd984894932d0b258ebbec1faecba74f/chips/stm32f4xx/src/trng.rs#L96): it simply accesses its wrapped `PeripheralClock` which itself implements the specific method. I also removed the `Option` from the return type of `get_frequency()`.